src/generators.rs

```diff
@@ -1072,6 +1072,393 @@ pub fn directed_binomial_tree_graph(
+
+    if d % 2 == 0 {
+        return Err(PyIndexError::new_err("d must be odd"));
+    }
+
+    let num_data = d * d;
+    let num_syndrome = (d - 1) * (d + 1) / 2;
+    let num_flag = d * (d - 1);
+
+    let nodes_data: Vec<NodeIndex> =
+        (0..num_data).map(|_| graph.add_node(py.None())).collect();
+    let nodes_syndrome: Vec<NodeIndex> = (0..num_syndrome)
+        .map(|_| graph.add_node(py.None()))
+        .collect();
+    let nodes_flag: Vec<NodeIndex> =
+        (0..num_flag).map(|_| graph.add_node(py.None())).collect();
+
+    // connect data and flags
+    for (i, flag_chunk) in nodes_flag.chunks(d - 1).enumerate() {
+        for (j, flag) in flag_chunk.iter().enumerate() {
+            graph.add_edge(nodes_data[i * d + j], *flag, py.None());
+            graph.add_edge(*flag, nodes_data[i * d + j + 1], py.None());
+            if bidirectional {
+                graph.add_edge(*flag, nodes_data[i * d + j], py.None());
+                graph.add_edge(nodes_data[i * d + j + 1], *flag, py.None());
+            }
+        }
+    }
+
+    // connect data and syndromes
+    for (i, syndrome_chunk) in nodes_syndrome.chunks((d + 1) / 2).enumerate() {
+        if i % 2 == 0 {
+            graph.add_edge(nodes_data[i * d], syndrome_chunk[0], py.None());
+            graph.add_edge(
+                syndrome_chunk[0],
+                nodes_data[(i + 1) * d],
+                py.None(),
+            );
+            if bidirectional {
+                graph.add_edge(syndrome_chunk[0], nodes_data[i * d], py.None());
+                graph.add_edge(
+                    nodes_data[(i + 1) * d],
+                    syndrome_chunk[0],
+                    py.None(),
+                );
+            }
+        } else if i % 2 == 1 {
+            graph.add_edge(
+                nodes_data[i * d + (d - 1)],
+                syndrome_chunk[syndrome_chunk.len() - 1],
+                py.None(),
+            );
+            graph.add_edge(
+                syndrome_chunk[syndrome_chunk.len() - 1],
+                nodes_data[i * d + (2 * d - 1)],
+                py.None(),
+            );
+            if bidirectional {
+                graph.add_edge(
+                    syndrome_chunk[syndrome_chunk.len() - 1],
+                    nodes_data[i * d + (d - 1)],
+                    py.None(),
+                );
+                graph.add_edge(
+                    nodes_data[i * d + (2 * d - 1)],
+                    syndrome_chunk[syndrome_chunk.len() - 1],
+                    py.None(),
+                );
+            }
+        }
+    }
+
+    // connect flag and syndromes
+    for (i, syndrome_chunk) in nodes_syndrome.chunks((d + 1) / 2).enumerate() {
+        if i % 2 == 0 {
+            for (j, syndrome) in syndrome_chunk.iter().enumerate() {
+                if j != 0 {
+                    graph.add_edge(
+                        nodes_flag[i * (d - 1) + 2 * (j - 1) + 1],
+                        *syndrome,
```

Comment:
Same here assuming I'm reading the figures in the paper correctly (the color change is annoying)

```suggestion
                        *syndrome,
                        nodes_flag[i * (d - 1) + 2 * (j - 1) + 1],
```